File: Data Structures & Algorithms/delete-node-in-a-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* deleteNode(TreeNode* root, int key) {
        if(!root){
            return nullptr;
        }

        if(key > root->val){
            root->right=deleteNode(root->right,key);
        }
        else if(key < root->val){
            root->left=deleteNode(root->left,key);
        }
        else{
            if(!root->left){
                return root->right;
            }
            else if(!root->right){
                return root->left;
            }
            else{
                TreeNode* minNode=findMinNode(root->right);
                root->val=minNode->val;
                root->right=deleteNode(root->right,minNode->val);
            }
        }
        return root;
    };

    TreeNode* findMinNode(TreeNode* root){
        TreeNode* curr=root;
        while (curr && curr->left){
            curr=curr->left;
        }
        return curr;
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        megaList=[]
        subset=[]
        queue=deque()
        if root:
            queue.append(root)
        level = 0
        while len(queue) > 0:
            subset=[]
            for i in range(len(queue)):
                curr = queue.popleft()
                #print(curr.val)
                subset.append(curr.val)
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
            level += 1
            megaList.append(subset)
        return megaList

        